src/main/java/io/vertx/core/CompositeFuture.java

```diff
@@ -246,4 +248,14 @@ default CompositeFuture onFailure(Handler<Throwable> handler) {
     }
     return list;
   }
+
+  /**
+   * @return a list of all the eventual failure causes. If no future failed, returns a list of null values.
+   */
+  @GenIgnore
+  default List<Throwable> causes() {
```

Comment:
can you instead do like list() do that creates less objects

Reply:
done